pkg/watcher/pv_watcher.go

```diff
@@ -159,18 +161,45 @@ func (pvw *PVWatcher) updateVolume(oldobj, obj interface{}) {
 			}
 		}()
 
-		ctxLogger.Info("Entry updateVolume()", zap.Reflect("obj", obj))
-		pv, _ := obj.(*v1.PersistentVolume)
+		ctxLogger.Info("Entry updateVolume()", zap.Reflect("obj", obj), zap.Reflect("oldobj", oldobj))
+		newpv, _ := obj.(*v1.PersistentVolume)
+		//If there is no change to status , capacity or iops we can skip the updateVolume call.
+		if oldobj != nil {
+			oldpv, _ := oldobj.(*v1.PersistentVolume)
+			oldCapacity := oldpv.Spec.Capacity[v1.ResourceStorage]
+			capacity := newpv.Spec.Capacity[v1.ResourceStorage]
+			iops := newpv.Spec.CSI.VolumeAttributes[utils.IOPSLabel]
+			oldiops := oldpv.Spec.CSI.VolumeAttributes[utils.IOPSLabel]
+
+			if (newpv.Status.Phase == oldpv.Status.Phase) && (oldCapacity.Value() == capacity.Value()) && (oldiops == iops) {
+				ctxLogger.Info("Skipping update Volume as there is no change in status , capacity and iops")
+				return
+			}
+		}
+
 		session, err := pvw.cloudProvider.GetProviderSession(context.Background(), ctxLogger)
 		if session != nil {
-			volume := pvw.getVolume(pv, ctxLogger)
-			ctxLogger.Info("volume to update ", zap.Reflect("volume", volume))
-			err := session.UpdateVolume(volume)
+			iksVpc, ok := session.(*iks_vpc_provider.IksVpcSession)
+
+			if !ok {
+				ctxLogger.Error("Failed to get the IKS-VPC session, Try to restart the CSI driver controller POD")
+				return
+			}
+
+			volume := pvw.getVolume(newpv, ctxLogger)
+			ctxLogger.Info("Updating metadata for the volume", zap.Reflect("volume", volume))
+			err := iksVpc.UpdateVolume(volume)
```

Comment:
please use iks session and then call UpdateVolume

Reply:
I had did that but later realised we have implemented this method as part of iksVPCSession object i.e iksVpc